Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -521,6 +521,24 @@ class BrowserViewController: UIViewController {
         log.debug("BVC calling super.viewDidAppear.")
         super.viewDidAppear(animated)
         log.debug("BVC done.")
+
+        let appVersion = AppInfo.appVersion
+        if let whatsNewURL = AppConstants.whatsNewURLForVersion(appVersion) where profile.prefs.stringForKey("latestAppVersion") != appVersion && hasConnectivity() {
```

Comment:
Try running this in a Playground:

```
func x() -> Int? {
    print("I was called")
    return 5
}

func y() -> Bool {
    print("So was I")
    return false
}

if let xx = x() where y() {
    print("Here")
}
```

to see what's up with this, then change it to

```
if profile.prefs.stringForKey(…) != appVersion && hasConnectivity() {
    self.openURLInNewTab(AppConstants.whatsN…(…))
}
```